web/src/app/components/header/header.component.ts

```diff
@@ -59,7 +66,26 @@ export class HeaderComponent {
   }
 
   onCancelEditSurveyClick() {
-    this.navigationService.selectSurvey(this.surveyId);
+    if (!this.draftSurveyService.dirty)
+      this.navigationService.selectSurvey(this.surveyId);
+    else {
```

Comment:
Let's use short-circuiting to avoid extra nesting, ie

```ts
 if (!this.draftSurveyService.dirty) {
      this.navigationService.selectSurvey(this.surveyId);
     return;
}
// .. do other stuff
```